### PR TITLE
Factory Orchestrator service runs on Linux as a standalone program

### LINKED ISSUE #99: Factory Orchestrator service runs on Linux as a standalone program
Let's ship a truly cross-platform task orchestration service!

This tracks shipping the service as a standalone program, not as a daemon. Future work will make it a daemon.

### PR DESCRIPTION
**Pull request by spaceisfun**
closes #99 

- Service runs on Linux as a standalone program when invoked with admin rights (sudo)
- Refactored service to store persistent state (through reboots) in an XML file
- Refactored service to store volatile state (lost after reboot), in XML file on Linux only. Windows still uses volatile registry.
- All service state modifications use FOServiceStatus & FOVolatileServiceStatus classes. Classes handle differences between OSes opaque to the service.
- Service configuration now uses appsettings.json file. Registry is still a fallback path on Windows, and allows OEM customizations to continue to work.
- Fixed issues on service preventing proper operation on Linux.
- Fixed issues with build scripts (can now use pwsh not only Windows PowerShell) & projects preventing proper operation on Linux. Everything but the app can be built on Linux without issue.
- Release build publishes Linux service zips
- Release build validates build & unit tests pass on Ubuntu
- Added new CommandLineTask class, for running scripts in bash/cmd. Identical to and Inherits from BatchFileTask to ensure back-compat.
- New GetOSPlatform() API
- App can connect to Linux service & functions as expected

### CODE REVIEW

File: StartClientSampleDebug.cmd [hunk 1]
```diff
@@ -1,2 +1,2 @@
 @echo off
-start cmd.exe /C "dotnet %~dp0\bin\Debug\AnyCPU\ClientSample\netcoreapp3.1\Microsoft.FactoryOrchestrator.ClientSample.dll %1 %2 %3 %4 & PAUSE"
\ No newline at end of file
+start cmd.exe /C "dotnet %~dp0bin\Debug\AnyCPU\Microsoft.FactoryOrchestrator.ClientSample\netcoreapp3.1\Microsoft.FactoryOrchestrator.ClientSample.dll %1 %2 %3 %4 & PAUSE"
```
Comment (hajya) [inline StartClientSampleDebug.cmd:2]: dropped a '\' after ~dp0, expected?

**Review by hajya — APPROVED** [review]
LGTM.. but tbh my brain hurt by about 3/4 through this.. if you eyes with scrutiny then i suggest setting up a meeting and walking folks through the changes. otherwise its hard to look at 64 files with variying changes and really feel like i gave it a solid review.

File: StartClientSampleDebug.cmd (hunk 1 above)
Comment (spaceisfun) [reply StartClientSampleDebug.cmd:2]: yeh dp0 includes a slash

**Pull request merged**